e2e/deployers/subscr.go

```diff
@@ -112,6 +112,16 @@ func (s Subscription) Undeploy(ctx types.TestContext) error {
 		return err
 	}
```

Comment:
Here we have 3 wait - with the current code each can run for 60 seconds, so we have unpredictable timeout depending on the number of object we wait for. This is not an issue if we use a deadline create before the waits and passed to the calls:

```go
deadline := time.Now().Add(3 * time.Minute)

if err := util.WaitPlacementDeleted(ctx, name, managementNamespace, deadline); err != nil {
    return err
}

if err := util.WaitManagedClusterSetBindingDeleted(ctx, name, managementNamespace, deadline); err != nil {
    return err
}

if err := util.WaitForNamespaceDelete(ctx, name, managementNamespace, deadline); err != nil {
    return err
}
```

To keep the changes small and easy to review, separate adding new infrastructure and code using it.